docker/install.sh

```diff
@@ -23,31 +23,61 @@ set -ex
 # ROS1 installation
 ###
 # Source the autoware installation
-source /opt/autoware.ai/ros/install/setup.bash
+
+if [[ ! -z "$ROS1_PACKAGES$ROS2_PACKAGES" ]]; then
+    echo "Sourcing previous build for incremental build start point..."
+    source /opt/carma/install/setup.bash
+else
+    echo "Sourcing base image for full build..."
+    source /opt/autoware.ai/ros/install/setup.bash
+fi
 
 cd ~/carma_ws
 
 sudo mkdir -p /opt/carma # Create install directory
 sudo chown carma /opt/carma # Set owner to expose permissions for build
 sudo chgrp carma /opt/carma # Set group to expose permissions for build
 
-echo "Building ROS1 CARMA Components"
-
-colcon build --install-base /opt/carma/install --cmake-args -DCMAKE_BUILD_TYPE=Release
-
+if [[ ! -z "$ROS1_PACKAGES$ROS2_PACKAGES" ]]; then
+    if [[ ! -z "$ROS1_PACKAGES" ]]; then
+        echo "Incrementally building ROS1 packages: $ROS1_PACKAGES"
+        colcon build --install-base /opt/carma/install --cmake-args -DCMAKE_BUILD_TYPE=Release --packages-above $ROS1_PACKAGES --allow-overriding $ROS1_PACKAGES
+    else
+        echo "Build type is incremental but no ROS1 packages specified, skipping ROS1 build..."
+    fi
+else
+    echo "Building all ROS1 CARMA Components"
+    colcon build --install-base /opt/carma/install --cmake-args -DCMAKE_BUILD_TYPE=Release
+fi
 echo "Build of ROS1 CARMA Components Complete"
 
 ###
 # ROS2 installation
 ###
 # Source the ROS2 autoware installation
 source /home/carma/catkin/setup.bash
-source /opt/autoware.ai/ros/install_ros2/setup.bash
+if [[ ! -z "$ROS1_PACKAGES$ROS2_PACKAGES" ]]; then
+    echo "Sourcing previous build for incremental build start point..."
+    source /opt/carma/install_ros2/setup.bash
+else
+    echo "Sourcing base image for full build..."
+    source /opt/autoware.ai/ros/install_ros2/setup.bash
+fi
 
 cd ~/carma_ws
 
 echo "Building ROS2 CARMA Components"
 
-colcon build --install-base /opt/carma/install_ros2 --build-base build_ros2 --cmake-args -DCMAKE_BUILD_TYPE=Release
+if [[ ! -z "$ROS1_PACKAGES$ROS2_PACKAGES" ]]; then
+    if [[ ! -z "$ROS2_PACKAGES" ]]; then
+        echo "Incrementally building ROS2 packages: $ROS2_PACKAGES"
+        colcon build --install-base /opt/carma/install_ros2 --cmake-args -DCMAKE_BUILD_TYPE=Release --packages-above $ROS2_PACKAGES --allow-overriding $ROS2_PACKAGES
```

Comment:
Probably good to specify the --build-base here as well

Reply:
Good catch, I missed that option!